tests/ga/test_remoteaccess_handler.py

```diff
@@ -14,21 +14,53 @@
 #
 # Requires Python 2.6+ and Openssl 1.0+
 #
-
 from datetime import timedelta, datetime
 
 from mock import Mock, MagicMock
-
-from azurelinuxagent.common.exception import RemoteAccessError
+from azurelinuxagent.common.osutil.default import DefaultOSUtil
 from azurelinuxagent.common.protocol.goal_state import RemoteAccess
 from azurelinuxagent.common.protocol.util import ProtocolUtil
 from azurelinuxagent.common.protocol.wire import WireProtocol
 from azurelinuxagent.ga.remoteaccess import RemoteAccessHandler
-from tests.common.osutil.mock_osutil import MockOSUtil
 from tests.tools import AgentTestCase, load_data, patch, clear_singleton_instances
-
-info_messages = []
-error_messages = []
+from tests.protocol.mocks import mock_wire_protocol
+from tests.protocol.mockwiredata import DATA_FILE, DATA_FILE_REMOTE_ACCESS
+
+
+class MockOSUtil(DefaultOSUtil):
+    def __init__(self):
+        self.all_users = {}
+        self.sudo_users = set()
+        self.jit_enabled = True
+
+    def useradd(self, username, expiration=None, comment=None):
+        if username == "":
+            raise Exception("test exception for bad username")
+        if username in self.all_users:
+            raise Exception("test exception, user already exists")
+        self.all_users[username] = (username, None, None, None, comment, None, None, expiration)
```

Comment:
Would a named tuple be better here (line 41)? I'm having trouble understanding what all the "None"s are.

Reply:
yes, probably. the existing tests need many improvements. i did several improvements as part of this pr and i may do more if i touch this code in the future.

Reply:
(btw i moved this mock from the common utils directory to this specific test, since its functionality is limited to the remote access tests)